docs/core/reports/customisation.md

```diff
@@ -83,6 +83,108 @@ Then this will be displayed at the top of reports:
 
 Note that you can also specify a path to a config file using `-c`.
 
+## Listing software versions
+
+For modules where version information can be extracted from log files,
+or when not available in log files, you can alternatively add them in
+the configuration. Software versions can be specified in YAML format
+using two options.
+
+### Option 1: Grouping software sersions by process name
+
+For each process, software versions can be organized within a nested YAML
+dictionary with three levels: `process` -> `software` -> `version(s)`.
+Multiple versions of the same software are specified as lists. Here's an
+example:
+
+```yaml
+software_versions:
+  quast:
+    quast:
+      - "5.2.0"
+      - "5.1.0"
+  samtools:
+    samtools: "1.11"
+    htslib: "1.3"
+  some_other_process:
+    some_other_tool:
+      - "2023-1"
+```
+
+:::warning
+Make sure that you write the version in quotes to ensure it being
+interpreted as a string. As an example a version `1.10` without
+quotes would be parsed as a float and displayed as version `1.1`.
+:::
+
+### Option 2: Dictionary of software name and version pairs
+
+An alternative is to specify pairs of software names and version
+strings directly. This option does not group software into processes.
+Instead the software name is used directly for the process name. Example:
+
+```yaml
+software_versions:
+  samblaster: "0.1.24"
+  quast:
+    - "5.2.0"
```

Comment:
I think we probably have _too much_ flexibility with the format now. Feels like you can throw basically whatever you want at this, which will probably be confusing. I was thinking before to force people to use one of two formats: simple or complex.

Simple is just name + version:

```yaml
software_versions:
  samblaster: "0.1.24"
```

Complex is the full way in the other direction:

```yml
software_versions:
  quast:
   - quast: "5.2.0"
   - quast: "5.1.0"
```

NB: List of dicts, which is actually not one of the formats you have here. But I think that this syntax is clearer / more flexible. If we have a list of dicts then there's not really a need for list of strings etc.

What do you think?

Reply:
I my opinion the list of dicts format is a bit repetitive. If you want to list multiple versions for a specific software, surely a list is cleaner. 

```yaml
software_versions:
  quast:
    quast: 
      - "5.2.0"
      - "5.1.0"
```

I also a bit confused about these two formats you suggest. 

In the *simple* format (that does not allow grouping) its software and version(s) is provided as a dict. Presumably this would also need to allow multiple versions so the format would be like this, not allowing single string inputs like above:

```yaml
software_versions:
  software:
    - version
    - other_version
```

In the *complex* format however, within each group you are instead passing a list of dicts specify versions.

```yaml
software_versions:
  group:
    - software: version
    - software: other_version
```

I think how you specify multiple versions should be the same if we choose to allow two formats. If we want to make the format a bit list flexible my suggestion would be these two formats. They would be similar to what we currently have but require a list of strings for each software. 

**1: Grouped input (complex)**
```yaml
software_versions:
  group:
    software: 
      - version
      - other_version
    other_software:
      - version
```

**2: Ungrouped input (simple)**
```yaml
software_versions:
  software: 
    - version
    - other_version
  other_software:
    - version
```

Reply:
Yup, you're right 👍🏻 

I can't remember the justification for the list of dicts now, which probably means it wasn't well thought through 😅  I think it was something along the lines of wanting to allow multiple identical keys - but there's no need for that if the key can have a list of versions. So yeah, I think I was just chasing my tail 😅 

Let's go for your two suggestions here 👍🏻

Reply:
I think this is how it's now implemented, right? So good to resolve 👍🏻

Reply:
Great!  

So my suggestion above is mostly how I implemented it except one thing. We currently allow single string input, we could possibly drop this to make it a bit more straightforward and a little less flexible. 

i.e. this single string 
```yaml
software_versions:
  software: version
```
would have to become a list
```yaml
software_versions:
  software: 
    - version
```

It would also mean less mixing of styles which would make it easier to read and write.

i.e. comparing something like this

```yaml
software_versions:
  software: 
    - version
  other_software: version
  other software_with_multiple_versions:
    - version1
    - version2
```

to this

```yaml
software_versions:
  software: 
    - version
  other_software: 
    - version
  other software_with_multiple_versions:
    - version1
    - version2
```

I think I am leaning for removing this input option.

Reply:
Just saw this... I don't really mind at this point, but I do want to get this merged 😆 

Whilst hypocritical compared to some stuff I've said recently, I'm happy enough. Let's leave as-is and merge.